src/Core/IdentityServer/BaseRequestValidator.cs

```diff
@@ -421,6 +424,10 @@ private Device GetDeviceFromRequest(ValidatedRequest request)
                             ["Challenges"] = tokens != null && tokens.Length > 1 ? tokens[1] : null
                         };
                     }
+                    else if (type == TwoFactorProviderType.WebAuthn)
+                    {
+                        return JsonSerializer.Deserialize<Dictionary<string, object>>(token);
```

Comment:
@Hinton , looks like this is now throwing `NullReferenceExceptions` on deserialization as premium members who cancel their subscription or it lapses now get a `null` token back and then get to this line and `throw`s.

Reply:
Solved in #1331